app/models/article.rb

```diff
@@ -73,4 +78,14 @@ def generate_published_dates
   def downcase_content_format
     self.content_format = self.content_format.downcase
   end
+
+  def create_redirect
+    unless Redirect.exists?(source_path: short_path, target_path: path) || short_path.blank?
+      Redirect.create(source_path: short_path, target_path: path)
```

Comment:
You'll prolly need to prepend `/` to `short_path` to make sure it matches up with the `Redirect.source_path` which all start with `/`.